### PR TITLE
Use field names from urlsplit() + Fix hardcoded shebangs

### PR DESCRIPTION
**Pull request by cclauss**
Also:
* Simplify exclusions
* Replace __re.match(sub, s)__ with __s.startswith(sub)__
* Replace __re.search(sub, s)__ with __sub in s__
* Fix shebang line to run with Python instead of legacy Python ;-)

### CODE REVIEW

**Comment by 0xInfection** [comment]
Can you explain why did you:
- Replace __re.match(sub, s)__ with __s.startswith(sub)__
- Replace __re.search(sub, s)__ with __sub in s__

And how did it help in the improving of the code. 🤔

**Comment by cclauss** [comment]
* I find match() and search() and findall() to be confusing...
* It is more intuitive for me to ask "Does s start with sub?" than to ask "Does s match sub?"
* It is more intuitive for me to ask "Is sub in s?" than to ask "Is sub in a search of s?"
* __in__ is useful across many Python data types, not just strings.
* __in__ is short and often used.
* If we did not need the __findall()__, we could have removed the import of __re__ altogether in favor of builtins.

File: modules/Uri_Checker.py [hunk 1]
```diff
@@ -10,43 +10,42 @@
...
+    if href == "http://localhost" or any(s in href for s in exclusions.split()):
         return '' # csrf stuff :o
 
-    parsed = urllib.parse.urlsplit(href) # :D
+    url_parts = urlsplit(url) # --> SplitResult(scheme, netloc, path, query, fragment)
+    href_parts = urlsplit(href)
     app = '' # init to url storage
 
-    if parsed[1] == urllib.parse.urlsplit(url)[1]:
+    if href_parts.netloc == url_parts.netloc:
```
Comment (0xInfection) [inline modules/Uri_Checker.py:27]: I really don't get why you did this. Lines 25-26. :(

> __Edit:__
>
> Got it. I wasn't aware of the fact that netlocs can be compared this way until I went through the docs once again. :) 
> +1 for the logic.

**Review by 0xInfection — COMMENTED** [review]
Parsing logic improved. Happy about that... but the change in `re` libraries are erring.

**Comment by 0xInfection** [comment]
About the usage of `re` library, I would say:
- `re.match` is anchored at the beginning of the string. That has nothing to do with newlines (`^`) which is the basic check of `.startswith()` attribute.
- We might use `in` in place of `re.search` since they both do the same work, but if the page has some Unicode, `in` might throw out exceptions. So I recommend using the default `re.search()`.

File: modules/Uri_Checker.py [hunk 2]
```diff
@@ -10,43 +10,43 @@
...
 import re
-from core.colors import * # import ends
+from urllib.parse import urlsplit
+from core.colors import O # import ends
 
 def buildUrl(url, href): # receive form input type / url
 
-    if re.search('logout',href) or re.search('action=out',href) or re.search('action=logoff', href) or re.search('action=delete',href) or re.search('UserLogout',href) or re.search('osCsid', href) or re.search('file_manager.php',href) or href=="http://localhost": # make exclusion list
+    exclusions = 'logout action=out action=logoff action=delete UserLogout osCsid file_manager.php'
+    if href == "http://localhost" or any(s in href for s in exclusions.split()):
```
Comment (0xInfection) [inline modules/Uri_Checker.py]:
```suggestion
    if href == "http://localhost" or any((re.search(s,href)) for s in exclusions.split()):
```

Comment (0xInfection) [reply modules/Uri_Checker.py]: You might consider this :D

File: modules/Uri_Checker.py [hunk 3]
```diff
@@ -10,43 +10,43 @@
...
+                app += '?' + href_parts.query # parameters :D
 
     return app
 
 def buildAction(url, action):
 
     print(O+'Parsing URL parameters...')
-    if action!='' and not re.match('#',action): # ;-; lets hope this stuff get what intended
-        return buildUrl(url,action) # get the url and reutrn it!
+    if action and not action.startswith('#'): # ;-; lets hope this stuff get what intended
```
Comment (0xInfection) [inline modules/Uri_Checker.py]: I would suggest reverting this change. :D

File: modules/Uri_Checker.py (hunk 2 above)
Comment (cclauss) [reply modules/Uri_Checker.py]: What values of __sub__ and what values of __s__ will break the last two lines in Python 3?
```python
sub = '#'
for s in ('#', '\n#', '#\n', '\n#\n'):
    assert bool(re.match('#', s)) == s.startswith('#')
    assert bool(re.search('#', s)) == ('#' in s)
```

Comment (0xInfection) [reply modules/Uri_Checker.py]: You have your answer [here](https://stackoverflow.com/questions/180986/what-is-the-difference-between-re-search-and-re-match). :)

Comment (cclauss) [reply modules/Uri_Checker.py]: Yes...  Only when you have a regex...  When you have normal strings (all str are Unicode in Py3) then the assert statements pass.

**Review by 0xInfection — APPROVED** [review]
Works as expected!

**Pull request merged**